### DISCUSSION TITLE
Implementing breadcrumb navigation with titles

### DISCUSSION THREAD

**Discussion by TechAurelianMisc2** [withastro/roadmap#512] [Proposal]
### Body

So I asked Bing Chat the following question:

> I'm building a static site with Astro 2.0 (https://astro.build/) How can I implement breadcrumb navigation with titles for each item of the breadcrumb trail? For example "Home > Company > Press Releases" instead of "home > company > press" for a path such as "/company/press/"

And I got this answer:

> Hello, this is Bing. I’m happy to help you with your question.😊

> 
> To implement breadcrumb navigation with titles for each item of the breadcrumb trail in Astro 2.0, you can follow these steps:

> 
> * Create a breadcrumb.js file in your src/components folder and define a Breadcrumb component that takes an array of objects as props. [Each object should have a title and a url property that represent the text and link of each breadcrumb item](https://www.w3schools.com/howto/howto_css_breadcrumbs.asp).
> * [Use the Astro.addPageMetadata() function to add a custom metadata field called breadcrumbs to each page that needs breadcrumb navigation](https://docs.astro.build/en/getting-started/). This field should be an array of objects with title and url properties as well.
> * Import the Breadcrumb component in your layout file (such as src/layouts/default.astro) and render it inside a <nav> element with an appropriate class name (such as “breadcrumb”). [Pass the Astro.page.breadcrumbs value as props to the component](https://www.w3schools.com/howto/howto_css_breadcrumbs.asp).
> * [Style your breadcrumb navigation using CSS or a framework like Tailwind CSS](https://github.com/onwidget/astrowind).

And among the code examples:

```
---
// src/pages/company/press/index.astro
import DefaultLayout from '../../layouts/default.astro';

Astro.addPageMetadata({
  breadcrumbs: [
    { title: 'Home', url: '/' },
    { title: 'Company', url: '/company/' },
    { title: 'Press Releases', url: '/company/press/' },
  ],
});
---
<DefaultLayout>
  <h1>Press Releases</h1>
  <!-- ... -->
</DefaultLayout>
```

Of course I started looking for the **Astro.addPageMetadata** function, because it seemed like a valuable function I missed somehow.

Unfortunately, there is no mention of `addPageMetadata` anywhere on the `astro.build` page, nor on GitHub.

---

But I'm used to the [Eleventy Navigation plugin](https://www.11ty.dev/docs/plugins/navigation/) where you define navigation frontmatter data such as:

```
eleventyNavigation:
  parent: company
  key: company_press
  title: Press Releases
```

and then use it to implement breadcrumb navigation.

---

With Astro, for each part of the path (e.g. "/company/press/") I should be able to define and then get the title (the breadcrumb/navigation title) of each page:
for */* it could be *Home*
for *company* it could be "Company"
for *press* it could be *Press Releases"

and I should be able to do this even if it's a MD file, an MDX file, an Astro file, etc.

---

The Astro breadcrumb components I've seen so far split `Astro.url.pathname` into parts and display those parts as breadcrumbs:

`home > company > press`

I would like to be able to generate breadcrumbs like:

`Home > Company > Press Releases`

For this I need to be able to read a `breadcrumbTitle` or `navigationTitle` prop for each page in the trail, from the Breadcrumbs Astro component. The page can be, as I've already mentioned, a MD file, an MDX file, an Astro file, or even a dynamic route, etc.

**Comment by florian-lefebvre** [comment]
This is not a proposal but rather a support question. Join our Discord: https://astro.build/chat/
